File: examples/src/main/python/simple_hive_context.py
from mist.mist_job import *

class SimpleHiveContext(MistJob, WithSQLSupport, WithHiveSupport):

    def execute(self, path):
        df = self.hive_context.read.json(path)
        df.printSchema()
        df.registerTempTable("people")

        df.show()
        df2 = self.hive_context.sql("SELECT AVG(age) AS avg_age FROM people")
        df2.show()

        result = df2.toJSON().first()

        return {"result": result}

File: examples/examples-spark1/src/main/scala/SimpleHiveContext.scala
import io.hydrosphere.mist.api.{HiveSupport, MistJob, SQLSupport}

object SimpleHiveContext extends MistJob with SQLSupport with HiveSupport {

  /** Contains implementation of spark job with [[org.apache.spark.sql.HiveContext]]
    * Abstract method must be overridden
    *
    * @param parameters user parameters
    * @return result of the job
    */
  def execute(file: String): Map[String, Any] = {
    val df = hiveContext.read.json(file)
    df.printSchema()
    df.registerTempTable("people")
    Map("result" -> hiveContext.sql("SELECT AVG(age) AS avg_age FROM people").collect())
  }
}
